File: 2nd Year/2nd Semester/MPP/PROIECT-JAVA/Laborator/TriatlonNetworking/src/main/java/triatlon/network/dto/TipProbaDTO.java
package triatlon.network.dto;

import java.io.Serializable;

public enum TipProbaDTO implements Serializable {
    NATATIE("Natatie"),
    CICLISM("Ciclism"),
    ALERGARE("Alergare");

    private final String denumire;

    TipProbaDTO(String denumire) {
        this.denumire = denumire;
    }

    public String getDenumire() {
        return denumire;
    }
}

File: 2nd Year/2nd Semester/MPP/Triatlon-C#/TriatlonNetworking/dto/TipProbaDTO.cs
namespace TriatlonNetworking.dto
{
    [Serializable]
    public enum TipProbaDTO
    {
        NATATIE,
        CICLISM,
        ALERGARE
    }

    // Extension methods for TipProbaDTO
    public static class TipProbaDTOExtensions
    {
        public static string GetDenumire(this TipProbaDTO tipProba)
        {
            return tipProba switch
            {
                TipProbaDTO.NATATIE => "Natatie",
                TipProbaDTO.CICLISM => "Ciclism",
                TipProbaDTO.ALERGARE => "Alergare",
                _ => throw new ArgumentException("Invalid TipProbaDTO value")
            };
        }
    }
}